coap-gateway/service/auth.go

```diff
@@ -68,3 +27,47 @@ func NewAuthInterceptor() Interceptor {
 		return ctx, expire.IsValid()
 	}
 }
+
+type jwtClaims jwt.MapClaims
+
+func (s *Service) ValidateToken(ctx context.Context, token string) (jwtClaims, error) {
+	ctx, cancel := context.WithTimeout(ctx, s.config.APIs.COAP.KeepAlive.Timeout)
+	defer cancel()
+	m, err := s.jwtValidator.ParseWithContext(ctx, token)
+	if err != nil {
+		return nil, err
+	}
+	return jwtClaims(m), nil
+}
+
+/// Get expiration time (exp) from user info map.
+/// It might not be set, in that case zero time and no error are returned.
+func (u jwtClaims) getExpirationTime() (time.Time, error) {
+	const expKey = "exp"
+	v, ok := u[expKey]
+	if !ok {
+		return time.Time{}, nil
+	}
+
+	exp, ok := v.(float64) // all integers are float64 in json
+	if !ok {
+		return time.Time{}, fmt.Errorf("invalid jwtClaims: invalid %v value type", expKey)
+	}
+	return pkgTime.Unix(int64(exp), 0), nil
+}
+
+/// Validate that ownerClaim is set and that it matches given user ID
+func (u jwtClaims) validateOwnerClaim(ocKey string, userID string) error {
+	v, ok := u[ocKey]
+	if !ok {
+		return fmt.Errorf("invalid jwtClaims: %v not set", ocKey)
+	}
+	ownerClaim, ok := v.(string)
+	if !ok {
+		return fmt.Errorf("invalid jwtClaims: %v", "invalid ownerClaim value type")
+	}
+	if ownerClaim != userID {
+		return fmt.Errorf("invalid ownerClaim: %v", userID)
```

Comment:
Again. owner claim is valid...

"owner identifier from the token '%v' doesn't match the one from the sign up message '%v'

Reply:
you are in package - you don't know who call you ... ` match the one from the sign up message '%v'`